include/dali/imgcodec/image_format.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2022, NVIDIA CORPORATION & AFFILIATES. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_IMGCODEC_IMAGE_FORMAT_H_
+#define DALI_IMGCODEC_IMAGE_FORMAT_H_
+
+#include <map>
+#include <memory>
+#include <stdexcept>
+#include <string>
+#include <vector>
+#include "dali/core/span.h"
+#include "dali/core/tensor_shape.h"
+#include "dali/core/stream.h"
+#include "dali/imgcodec/image_source.h"
+
+namespace dali {
+namespace imgcodec {
+
+struct ImageInfo {
+  TensorShape<> shape;
+  struct {
+    int rotate;
+    bool flip_x;
+    bool flip_y;
+  } orientation;
+};
+
+class ImageParser {
+ public:
+  virtual ~ImageParser() = default;
+
+  /**
+   * @brief   Parses the encoded image to get image information (shape, ...)
+   * @remarks ImageInfo will be valid only if `CanParse` returned true
+   */
+  virtual ImageInfo GetInfo(ImageSource *encoded) const = 0;
+
+  /**
+   * @brief Verifies whether the parser can understand an encoded image,
+   *        that is, if it's in the format that this parser handles.
+   */
+  virtual bool CanParse(ImageSource *encoded) const = 0;
+};
+
+class ImageDecoder;
+
+class DLL_PUBLIC ImageFormat {
```

Comment:
One slightly confusing thing naming-wise: there are ImageParser and ImageFormat classes and the directory called formats which contains several ImageParsers. :)

Reply:
Renamed to `parsers`